core/src/test/scala/integration/kafka/api/ProducerCompressionTest.scala

```diff
@@ -103,12 +123,37 @@ class ProducerCompressionTest extends QuorumTestHarness {
       // make sure the fetched message count match
       consumer.assign(Collections.singleton(tp))
       consumer.seek(tp, 0)
-      val records = TestUtils.consumeRecords(consumer, numRecords)
+      val records = TestUtils.consumeRecords(consumer, numRecords*3)
+
+      for (i <- 0 until numRecords) {
+        val messageValue = messageValues.apply(i)
```

Comment:
This is a heavy nitpick, but usually `apply` is not spelled in Scala. This should read:
```suggestion
        val messageValue = messageValues(i)
```